src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -501,148 +1387,699 @@ private boolean isAmqpHeaderValid(
+            {
+                creditor.release(replyBudgetIndex);
+                replyBudgetIndex = NO_CREDITOR_INDEX;
+            }
         }
 
-        private void onAmqpFlow(
-            AmqpFrameFW flow)
+        private void cleanupDecodeSlotIfNecessary()
         {
-            // TODO
+            if (decodeSlot != NO_SLOT)
+            {
+                bufferPool.release(decodeSlot);
+                decodeSlot = NO_SLOT;
+                decodeSlotLimit = 0;
+            }
         }
 
-        private void onAmqpTransfer(
-            AmqpFrameFW transfer)
+        private void cleanupEncodeSlotIfNecessary()
         {
-            // TODO
+            if (encodeSlot != NO_SLOT)
+            {
+                bufferPool.release(encodeSlot);
+                encodeSlot = NO_SLOT;
+                encodeSlotOffset = 0;
+                encodeSlotTraceId = 0;
+            }
         }
 
-        private void onAmqpDisposition(
-            AmqpFrameFW disposition)
+        private final class AmqpSession
         {
-            // TODO
+            private AmqpSession()
+            {
+                // TODO
+            }
+
+            private void onDecodeBegin(
+                long traceId,
+                long authorization)
+            {
+                doEncodeBegin(traceId, authorization);
+            }
+
+            private void onDecodeAttach(
+                long traceId,
+                long authorization,
+                AmqpAttachFW attach)
+            {
+                final int linkNameKey = linkNameKey(attach.name().asString());
+                if (links.containsKey(linkNameKey))
+                {
+                    onDecodeError(traceId, authorization, NOT_ALLOWED);
+                }
+                AmqpServerStream link = links.computeIfAbsent(linkNameKey, l -> new AmqpServerStream());
+                link.onDecodeAttach(traceId, authorization, attach);
+            }
         }
 
-        private void onAmqpDetach(
-            AmqpFrameFW detach)
+        private class AmqpServerStream
         {
-            // TODO
+            private MessageConsumer application;
+            private long newRouteId;
+            private long initialId;
+            private long replyId;
+            private long budgetId;
+            private int replyBudget;
+            private int sharedReplyBudget;
+
+            private int initialSlot = NO_SLOT;
+            private int initialSlotOffset;
+            private long initialSlotTraceId;
+
+            private int state;
+            private Set<AmqpRole> roles;
+
+            AmqpServerStream()
+            {
+                this.roles = EnumSet.noneOf(AmqpRole.class);
+            }
+
+            private void onDecodeAttach(
+                long traceId,
+                long authorization,
+                AmqpAttachFW attach)
+            {
+                linkName = attach.name().asString();
+                handle = attach.handle();
```

Comment:
Use `this.` for assignments like `this.linkName` and `this.handle` to make it clear we're updating instance state.
Also `linkName` can just be `name`.